### DISCUSSION TITLE
FIFO Queue vs Dropping lowest priority

### DISCUSSION THREAD

**Discussion by WilderLavington** [ymd-h/cpprb#26] [General]
Hello, I am currently using the cpprb.PrioritizedReplayBuffer, and was wondering if it is possible to set how the buffer handles when you have reached more examples then the max buffer-size set. Currently it seems like it always follows the usual FIFO setup, however it would be nice if we could drop based upon priority (even if the user has to do it manually). Is there currently a hyper-parameter I can set or some function I can call which will drop a specific set of indices from the current buffer?  Any solution you can think of would be super helpful!

**Comment by ymd-h** [comment]
@WilderLavington 

Thank you for your feedback.

> Is there currently a hyper-parameter I can set or some function I can call which will drop a specific set of indices from the current buffer?

No, all `cpprb` buffers follow FIFO setup for dropping (or overwriting), and users cannot customize it.

I know there is a need to drop item more flexibly, and such feature is implemented at another project like [DeepMind's Reverve](https://github.com/deepmind/reverb), however, there is a difficulty that some features like [memory compression](https://ymd_h.gitlab.io/cpprb/features/memory_compression/) and [multiprocessing](https://ymd_h.gitlab.io/cpprb/features/ape-x/) rely on assumption of sequential data storing.

I have no idea to solve this difficulty yet.
